src/main/resources/MessagingDescriptor.json

```diff
@@ -5,8 +5,14 @@
       "description": "Patron fee/fine balance was changed",
       "eventTTL": 1,
       "signed": false
+    },
+    {
+      "eventType": "FF_ACCOUNT_WITH_LOAN_CLOSED",
```

Comment:
We are already issueing an event `FF_BALANCE_CHANGED` when an account is closed. Can we reuse that for when an account related to a loan is closed? We can add `loanId` to the existing event's payload.

Reply:
Hmm... That's a good point.

The idea in publishing this event is to be able close the associated loan when all `lost item fees` are closed/resolved.

So, mod-circulation will subscribe to this event type. I'm afraid that this will overuse circulation computing resources, because the `FF_BALANCE_CHANGED` event will be generated when an account (either automated|manual) created/updated (when paid/waived/transferred partially|fully)/deleted. And for each generated event we will have to:
# Fetch the loan;
# Fetch all associated accounts.

By introducing the `FF_ACCOUNT_WITH_LOAN_CLOSED` event, we minimize amount of events, so we'll receive most likely events that we're interesting in  (closed and automated).

Does it make sense?

Reply:
You are right, events for closed "acccounts" related to loans are just a small subset of events of type `FF_BALANCE_CHANGE` issued by mod-feesfines.

Reply:
Is it too late to change the name of the event to `FEEFINE_WITH_LOAN_CLOSED`? A major refactoring of mod-feesfines is planned, which will involve revision of terminology used by the module. So the term `account` is likely to be replaced with something that makes a bit more sense.

Reply:
> Is it too late to change the name of the event to `FEEFINE_WITH_LOAN_CLOSED`? 

What would you suggest as an alternative?

> A major refactoring of mod-feesfines is planned, which will involve revision of terminology used by the module. So the term `account` is likely to be replaced with something that makes a bit more sense.

I'm glad that this is being done. Please can you refer me to further information about this e.g. a JIRA issue, planned domain model etc.

Reply:
@OleksandrVidinieiev, `FEEFINE_WITH_LOAN_CLOSED` is fine for me, will replace. Thanks.

Reply:
> What would you suggest as an alternative?
This is the alternative. The original name was `FF_ACCOUNT_WITH_LOAN_CLOSED`.

Reply:
@OleksandrVidinieiev Does that mean that the new terminology is going to be `fee / fine` rather than `account`?

Reply:
@marcjohnson-kint The new terminology does not exist yet. But if we change the term `account` to something else, event `FEEFINE_ACCOUNT_WITH_LOAN_CLOSED` will definitely need to be renamed too. On the other hand, stripping this name of the `ACCOUNT` part now will:
a) preserve its meaning: account is an instance of a fee/fine, closing a fee/fine means closing an account
b) eliminate the need to rename the event after `account` is renamed to something else

Does it make sense?

Reply:
@OleksandrVidinieiev Thanks for explaining that more.

I'm ok with removing the account aspect now, and then reviewing if the names need to change further when the domain analysis has been done.

(The other day the term `bill` was mentioned to me for referring to what is issued when a fee or fine is charged to a patron)

Reply:
@marcjohnson-kint I've heard this entity being referred to as _bill_ too. If that's an established term, it makes it a likely candidate for _account_ replacement.

Reply:
> @marcjohnson-kint I've heard this entity being referred to as _bill_ too. If that's an established term, it makes it a likely candidate for _account_ replacement.

I don't know if it is an established term. I heard it for the first time last week. I'd like it to be considered during the analysis of the domain.